arrow/src/compute/kernels/sort.rs

```diff
@@ -480,18 +480,20 @@ where
     let mut nulls = null_indices;
 
     let valids_len = valids.len();
-    let nulls_len = nulls.len();
+    let mut nulls_len = nulls.len();
     let mut len = values.len();
 
     if let Some(limit) = limit {
         len = limit.min(len);
+        // count how many nulls are present in the limit
+        nulls_len = nulls.iter().filter(|&idx| *idx as usize <= limit).count();
```

Comment:
Everything else in this function doesn't need to dig into the actual index values so something smells here to me. I thought the limit is applied to the sorted result so I don't know that comparing the limit to the indices is correct here.